pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -898,10 +898,12 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 	// Now validate all the listeners. Collate the tcp listeners first and then the HTTP listeners
 	// TODO: This is going to be bad for caching as the order of listeners in tcpListeners or httpListeners is not
 	// guaranteed.
+	invalid := 0.0
 	for name, l := range listenerMap {
 		if err := l.listener.Validate(); err != nil {
 			log.Warnf("buildSidecarOutboundListeners: error validating listener %s (type %v): %v", name, l.servicePort.Protocol, err)
-			invalidOutboundListeners.Add(1)
+			invalid++
+			invalidOutboundListeners.Record(invalid)
```

Comment:
It seems before this was effectively a counter so this changes the behavior -- I think this is good though. I think later in the code if we find a valid one we panic anyways ref: https://github.com/istio/istio/issues/14744. Will follow up on this in #14744.

Reply:
yeah, the use as a `Gauge` was potentially misleading. OC's `LastValue` acts differently (there isn't really a "take current value and add one to it, but also allow it to reset to zero, and export it as a gauge" equivalent). This is something we could patch if/as we evolve the monitoring API.